sg_otio/sg_settings.py

```diff
@@ -410,15 +410,23 @@ def __init__(self, sg, linked_entity_type=None):
         self.validate_shot_cut_fields_prefix(sg_settings.shot_cut_fields_prefix)
         self._shot_cut_fields_prefix = sg_settings.shot_cut_fields_prefix
         self._sg_shot_link_field = None
-        self._shot_schema = None
+        self.cache_shot_schema()
+
+    def cache_shot_schema(self):
+        """
+        Cache the schema for the Shot entity, if possible.
+        """
+        if self._sg and self._shot_schema is None:
+            # For some reason, setting self._shot_schema does not set it at the
```

Comment:
Should this method be a `classmethod`?

Reply:
if it was, sg handle should be either provided to it or switched from an instance attribute to a class attribute